### ISSUE TITLE
synchronous call for client

### ISSUE THREAD

**Issue by receiver1** [qchateau/packio#82]
Hello library developer! I ask you to please add a synchronous call for the client. Asynchrony is not possible in all environments, so this will be very useful and I think not only for me.

**Comment by qchateau** [comment]
Hello, what upside so you see in having a blocking API while you can just `async_call` then run your `io_context` ?
Simplicity would obviously be one but an adapter would do to hide that

**Comment by receiver1** [comment]
> Hello, what upside so you see in having a blocking API while you can just `async_call` then run your `io_context` ? Simplicity would obviously be one but an adapter would do to hide that

I use asio with 70% of its functionality disabled because in my environment the use of streams is prohibited and unavailable.

**Comment by qchateau** [comment]
Async doesn't need multiple threads. I don't know what you mean by "70% of asio disabled" but if you mean you restrict yourself to a subset of the features available then I can't accommodate such request as that set would be different for every potential user

**Comment by receiver1** [comment]
> Async doesn't need multiple threads, on the contrary it is most useful in single threaded contexts. I don't know what you mean by "70% of asio disabled" but if you mean you restrict yourself to a subset of the features available then I can't accommodate such request as that set would be different for every potential user

No, I mean that in Boost and Asio there are settings via defines. There is such a define as BOOST_DISABLE_THREADS, which disables threads, and accordingly all asynchronous calls. It would be very kind of you to implement a synchronous call on the client, since asynchronicity and threads are not supported in every runtime.

**Comment by qchateau** [comment]
Thanks for the context. Is there any other configuration macro you are interested in apart from ASIO_DISABLE_THREADS ?
